packages/preview/progressive-layout/1.0.0/README.md

```diff
@@ -0,0 +1,68 @@
+# progressive-outline
+
+This package provides a Typst function, `progressive-outline`, to generate a dynamic table of contents, similar to what is found in some LaTeX's Beamer presentations.
+
+## Context and Ultimate Goal
+
+This function was developed to meet a need within the Typst community for displaying partial or progressive outlines, especially in presentations. It is inspired by the following discussions:
+
+-   [GitHub Issue (touying)](https://github.com/touying-typ/touying/issues/137)
+-   [Typst Forum](https://forum.typst.app/t/how-to-display-partial-outlines-with-touying/5526)
+
+The ultimate goal would be to integrate this functionality directly into a presentation package like **[`polylux`](https://typst.app/universe/package/polylux/)**, **[`touying`](https://typst.app/universe/package/touying/)**, or **[`presentate`](https://typst.app/universe/package/presentate/)**. This would allow for the creation of "Beamer-like" presentations where the outline unfolds flexibly as you progress through the slides. Feel free to draw inspiration from this code to contribute to these projects!
+
+## Function Usage
+
+To use the function, import the package and call `progressive-outline` where you want the outline to appear.
+
+```typst
+#import "@preview/progressive-layout:1.0.0": progressive-outline
+
+#progressive-outline(
+  h1-style: "all",
+  h2-style: "current-and-grayed",
+  h3-style: "current",
+  show-numbering: true,
+)
+```
+
+## Example: The `progressive-layout` Template
+
+This package also contains a simple template, `progressive-layout`, which shows an example of how to use the function. Here is a minimal code example to test it:
+
+```typst
+// Import the template
+#import "@preview/progressive-layout:1.0.0": progressive-layout, progressive-outline
+
+// Apply the template to your document
+#show: doc => progressive-layout(doc)
+
+// Your headings...
+= Chapter 1
+== Section 1.1
+= Chapter 2
+```
+
+## Examples
+
+This package includes two files to demonstrate its usage:
+
+-   [`example.typ`](https://github.com/eusebe/progressive-layout/blob/main/example.typ): This file shows how the `progressive-outline` function works with different options.
+-   [`demo.typ`](https://github.com/eusebe/progressive-layout/blob/main/demo.typ): This file provides an example of how to use the `progressive-layout` template.
```

Comment:
I would either:
- link to the files in this repository (i.e. `[\`example.typ\`](example.typ)`)
- delete the copy of the files in this repository, and link to a pinned commit or tag in your own repository (by replacing `main` in the URL with the commit hash or tag name)
This was, you can be sure that the example files will always be in sync with the version of the package the user is reading the README for. Currently, if you push a breaking API change, the examples won't work anymore for people still reading the old README.